Workflow license and creator edit keyboard access

itisAliRH opened the pull request:
Closes #18915. This PR improves the `LicenseSelector` and `CreatorEditor` ui and makes them keyboard accessible. It migrates `LicenseSelector` to composition API and typescript, uses GalaxyAPI to fetch license options, and uses the `MultiSelect` component, simplifying the license handling steps.

https://github.com/user-attachments/assets/6efa07a2-ae2d-4056-b640-2deac21b968e

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. In the workflow editor, access and edit license and creator via keyboard

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

itisAliRH committed:
🛠️: migrate `LicenseSelector` to the composition API and typescript, add error alert, improve button controls and make them accessible via keyboard

itisAliRH committed:
🛠️: make edit and cancel buttons in `CreatorEditor` accessible via keyboard

mvdbeek commented on client/src/components/License/LicenseSelector.vue:
```diff
@@ -1,130 +1,161 @@
+    }
+);
+
+onMounted(async () => {
```
Why `onMounted` ? There's no need for that. Use `onMounted` only if you need the DOM before making a request.

itisAliRH committed:
🛠️: replace unnecessary `onMounted` in `LicenseSelector` with an async function

itisAliRH committed:
Redesign license selection using `vue-multiselect` for improved usability

Updated the `LicenseSelector` component to enhance user experience by replace a `Multiselect` form `vue-multiselect` dropdown and simplifying the license handling steps and logic.

ElectronicBlueberry commented on client/src/components/License/LicenseSelector.vue:
```diff
@@ -1,130 +1,111 @@
+    }
+);
+
+setCurrentLicense();
```
use an immediate watcher here instead

ElectronicBlueberry commented on client/src/components/License/LicenseSelector.vue:
```diff
@@ -1,130 +1,111 @@
+async function setCurrentLicense() {
+    if (!licenses.value?.length) {
+        await fetchLicenses();
+    }
```
this feels like it could easily trigger a ton of requests if something goes wrong. Can you add some sort of loading block + indicator?

ElectronicBlueberry commented on client/src/components/License/LicenseSelector.vue:
```diff
@@ -1,130 +1,111 @@
+        errorMessage.value = errorMessageAsString(error) || "Unable to fetch licenses.";
+    }
+
+    licenses.value = data;
```
Could we put this in a store, so it doesn't reload on re-rendering this component?

itisAliRH committed:
Updates workflow editor license selection

Refactors license selection logic in the workflow editor
Simplifies selectors and improves interaction handling
Enhances test assertions for license value verification

itisAliRH committed:
Fix license loading logic and improve reactivity

Changes initial licensesLoading state to false
Uses watchImmediate for immediate reactivity on inputLicense
Ensures licenses are fetched only when not already loading
